internal/flags/apiflags/flags_test.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiflags
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAddTo(t *testing.T) {
+	testCases := []struct {
+		name     string
+		apiFlags APIFlags
+		validate func(APIFlags, *pflag.FlagSet) error
+	}{
+		{
+			// Populate FlagSet
+			name: "Populate FlagSet",
+			apiFlags: APIFlags{
+				APIVersion:       "app.example.com/v1alpha1",
+				Kind:             "AppService",
+				SkipGeneration:   false,
+				CrdVersion:       "v1",
+				HelmChartRef:     "stable/app",
+				HelmChartVersion: "1.0.0",
+				HelmChartRepo:    "https://charts.mycompany.com/",
+			},
+			validate: func(apiFlags APIFlags, flagSet *pflag.FlagSet) error {
+				val, err := flagSet.GetString("api-version")
+				if err != nil {
+					return err
+				}
+				if apiFlags.APIVersion != val {
+					return fmt.Errorf("apiVersion does not match")
```

Comment:
Why the errors has been returned in the tests?
It shows not correct. What it is actually testing? 
The errors should be returned by the implementation and not be fixed in the tests. 
c/c @joelanford

Reply:
@camilamacedo86 because this is a validation method for the test. The `AddTo` method does NOT return errors. It simply sets values on the `FlagSet`. So the validation function is validating that the `FlagSet` received the values correctly from the `APIFlags`. If they do not match the expected values, we return an error so that the test will fail.